tools/driver/swift_format_main.cpp

```diff
@@ -0,0 +1,295 @@
+                       diag::error_formatting_invalid_range);
+        return true;
+      }
+      if (FromLine > ToLine) {
+        Diags.diagnose(SourceLoc(),
+                       diag::error_formatting_invalid_range);
+        return true;
+      }
+      for (unsigned Line = FromLine; Line <= ToLine; ++Line) {
+        size_t Offset = getOffsetOfLine(Line, Output);
+        ssize_t Length = getOffsetOfLine(Line + 1, Output) - 1 - Offset;
+        if (Length < 0)
+          break;
+
+        CodeFormatOptions FormatOptions;
+        FormatOptions.UseTabs = UseTabs;
+        FormatOptions.IndentWidth = IndentWidth;
+        FormatOptions.TabWidth = TabWidth;
+        std::string Formatted = Doc.reformat(LineRange(Line, 1),
+                                             FormatOptions).second;
+        if (Formatted.find_first_not_of(" \t\v\f", 0) == StringRef::npos)
+          Formatted = "";
+
+        if (Formatted == Output.substr(Offset, Length))
+          continue;
+
+        Output.replace(Offset, Length, Formatted);
+        Doc.updateCode(llvm::MemoryBuffer::getMemBuffer(Output));
+        Replacements.insert(clang::tooling::Replacement(Filename,
+                                                        Offset,
+                                                        Length,
+                                                        Formatted));
+      }
+      if ((InPlace && Filename == "-") || OutputFilename == "-") {
+        llvm::outs() << Output;
+        return false;
+      }
+      std::error_code EC;
+      StringRef Destination;
+      if (InPlace)
+        Destination = Filename;
+      else
+        Destination = OutputFilename;
+      llvm::raw_fd_ostream out(Destination, EC, llvm::sys::fs::F_None);
+      if (out.has_error() || EC) {
+        Diags.diagnose(SourceLoc(),
+                       diag::error_opening_output,
+                       Filename,
+                       EC.message());
+        out.clear_error();
+        return true;
+      }
+      out << Output;
+    }
+    return false;
+  }
+};
+
+int swift_format_main(ArrayRef<const char *> Args, const char *Argv0,
+                      void *MainAddr) {
+  CompilerInstance Instance;
+  PrintingDiagnosticConsumer PDC;
+  Instance.addDiagnosticConsumer(&PDC);
+
+  SwiftFormatInvocation Invocation;
+  std::string MainExecutablePath = llvm::sys::fs::getMainExecutable(Argv0,
+                                                                    MainAddr);
+  Invocation.setMainExecutablePath(MainExecutablePath);
+
+  DiagnosticEngine &Diags = Instance.getDiags();
+  if (Invocation.parseArgs(Args, Diags) != 0)
+    return 1;
+
+  std::vector<std::string> InputFiles = Invocation.getInputFilenames();
+  unsigned NumInputFiles = InputFiles.size();
+  if (NumInputFiles == 0)
+    // Read source code from standard input.
+    Invocation.format("-", Diags);
+  else if (NumInputFiles == 1)
+    Invocation.format(InputFiles[0], Diags);
+  else {
```

Comment:
Please use braces around the if/else if/else here ... it tends to make it less readable if some of the blocks are braced and some are not.